specification/storageactions/StorageAction.Management/StorageTask.tsp

```diff
@@ -0,0 +1,145 @@
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.StorageActions;
+/**
+ * Represents Storage Task.
+ */
+@Azure.ResourceManager.Private.armResourceInternal(StorageTaskProperties)
+@Http.Private.includeInapplicableMetadataInPayload(false)
+model StorageTask extends Foundations.TrackedResource {
+  ...ResourceNameParameter<
+    Resource = StorageTask,
+    KeyName = "storageTaskName",
+    SegmentName = "storageTasks",
+    NamePattern = "^[a-z0-9]{3,18}$"
+  >;
+
+  /**
+   * The managed service identity of the resource.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "For backward compatibility"
+  identity: Azure.ResourceManager.Foundations.ManagedServiceIdentity;
+
+  @doc("The resource-specific properties for this resource.")
+  @Azure.ResourceManager.Private.conditionalClientFlatten
+  @Azure.ResourceManager.Private.armResourcePropertiesOptionality(false)
+  properties?: StorageTaskProperties;
+}
+
+@armResourceOperations
+interface StorageTasks {
+  /**
+   * Get the storage task properties
+   */
+  get is ArmResourceRead<StorageTask>;
+
+  /**
+   * Asynchronously creates a new storage task resource with the specified parameters. If a storage task is already created and a subsequent create request is issued with different properties, the storage task properties will be updated. If a storage task is already created and a subsequent create or update request is issued with the exact same set of properties, the request will succeed.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-put-operation-response-codes" "For backward compatibility"
+  create is ArmResourceCreateOrReplaceAsync<
+    StorageTask,
+    Response = ArmResourceUpdatedResponse<StorageTask> | ArmResourceCreatedResponse<
+      StorageTask,
+      ArmLroLocationHeader & Azure.Core.Foundations.RetryAfterHeader
+    > | ArmAcceptedLroResponse
+  >;
+
+  /**
+   * Update storage task properties
+   */
+  @patch(#{ implicitOptionality: false })
+  update is ArmCustomPatchAsync<
+    StorageTask,
+    PatchModel = StorageTaskUpdateParameters
+  >;
+
+  /**
+   * Delete the storage task resource.
+   */
+  delete is ArmResourceDeleteWithoutOkAsync<StorageTask>;
+
+  /**
+   * Lists all the storage tasks available under the given resource group.
+   */
+  listByResourceGroup is ArmResourceListByParent<
+    StorageTask,
+    Response = StorageTasksListResult
+  >;
+
+  /**
+   * Lists all the storage tasks available under the subscription.
+   */
+  listBySubscription is ArmListBySubscription<
+    StorageTask,
+    Response = StorageTasksListResult
+  >;
+
+  /**
+   * Fetch the storage tasks run report summary for each assignment.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @get
+  @action("reports")
+  @operationId("StorageTasksReport_List")
```

Comment:
These `@operationId` seems avoidable?

One can put the op in `StorageTasksReport` interface, and do `@clientName("list")` on it (well in this case op name is same).

Reply:
I don't think we should rush to your suggestion until we have a mature design on deprecating operationId. I insist operations on the same resource should group together by default.